spec/jobs/emit_web_hook_event_spec.rb

```diff
@@ -24,6 +24,47 @@
     end.to raise_error(Discourse::InvalidParameters)
   end
 
+  context 'when the web hook is failed' do
+    before do
+      SiteSetting.retry_web_hook_events = true
+      stub_request(:post, "https://meta.discourse.org/webhook_listener")
+        .to_return(body: 'Invalid Access', status: 403)
+    end
+
+    it 'retry if site setting is enabled' do
+      expect(Jobs::EmitWebHookEvent.jobs.size).to eq(0)
+      subject.execute(
+        web_hook_id: post_hook.id,
+        event_type: described_class::PING_EVENT
+      )
+      expect(Jobs::EmitWebHookEvent.jobs.size).to eq(1)
+      job = Jobs::EmitWebHookEvent.jobs.first
+      args = job["args"].first
+      expect(args["retry_count"]).to eq(1)
+    end
+
+    it 'does not retry for more than 4 times' do
```

Comment:
This is related to my comment about magic numbers because if we change the number in the sidekiq job, this description will no longer be accurate.